File: Week 17/Lecture15/BankAcct/BankAcct.py
class BankAcct(object):
    def __init__(self, *arg): # aNum, bal
        if len(arg)==0:
            self.acctNum = ""
            self.balance = 0
        elif len(arg)==1:
            self.acctNum = arg[0]
            self.balance = 0
        else:
            self.acctNum = arg[0]
            self.balance = arg[1]

    def withdraw(self, amount):
        if self.balance >= amount:
            self.balance -= amount
            return True
        else:
            return False

    def deposit(self, amount):
        if (amount <= 0):
            return "Can't deposit non-positive amount"
        self.balance += amount
        return 

    def print(self):
        print("Account Number:", self.acctNum)
        print("Balance:", self.balance)

def transfer(fromAcct, toAcct, amount):
    if fromAcct.withdraw(amount):
        toAcct.deposit(amount)
    else:
        print("Money not enough")

ba1 = BankAcct()
ba2 = BankAcct(1234, 99.99)

ba1.deposit( 1000 );
ba2.withdraw( 50.25 );     
    
transfer( ba1, ba2, 50.0);
print("printing ba1");
ba1.print();
    
print("printing ba2");
ba2.print();

transfer( ba2, ba1, 100)
print("printing ba1");
ba1.print();
    
print("printing ba2");
ba2.print();
    

File: Week 17/BankAcct/src/BankAcct.java
class BankAcct
{
  protected int _acctNum;
  protected double _balance; 
  //Constructors
  public BankAcct( ) {   
    //initialize all attributes to 0
  }
  public BankAcct( int aNum, double bal ) {
    //initialize attributes with user provided values
    _acctNum = aNum;
    _balance = bal;
  }
  
  //Methods
  public boolean withdraw( double amount ) {   
    if ( _balance < amount )
      return false;    
    _balance -= amount;
    return true;            
  }

  public void deposit( double amount ) {
    if ( amount <= 0 )
      return; 
    _balance += amount;
  }
 
  public void print() {
    System.out.println( "Account Number: " + _acctNum );
    System.out.printf( "Balance: $%.2f\n", _balance );
  }
}

